src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -2063,6 +2063,7 @@ DataPartStoragePtr IMergeTreeDataPart::makeCloneInDetached(const String & prefix
     IDataPartStorage::ClonePartParams params
     {
         .copy_instead_of_hardlink = isStoredOnRemoteDiskWithZeroCopySupport() && storage.supportsReplication() && storage_settings->allow_remote_fs_zero_copy_replication,
+        .keep_metadata_version = prefix == "covered-by-broken",
```

Comment:
Nice catch, thank you! Another possible solution is excluding files like "metadata version" from `DataPartStorageOnDiskBase::looksLikeBrokenDetachedPartHasTheSameContent`